src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -1163,34 +1166,56 @@ void addSubpatternsForTuple(ImmutableArray<TypeWithAnnotations> elementTypes)
             var builder = ArrayBuilder<BoundSubpattern>.GetInstance(node.Subpatterns.Count);
             foreach (SubpatternSyntax p in node.Subpatterns)
             {
-                IdentifierNameSyntax? name = p.NameColon?.Name;
+                ExpressionSyntax? expr = p.ExpressionColon?.Expression;
                 PatternSyntax pattern = p.Pattern;
-                Symbol? member = null;
+                ImmutableArray<Symbol> members;
                 TypeSymbol memberType;
-                if (name == null)
+                if (expr == null)
                 {
                     if (!hasErrors)
                         diagnostics.Add(ErrorCode.ERR_PropertyPatternNameMissing, pattern.Location, pattern);
 
                     memberType = CreateErrorType();
+                    members = ImmutableArray<Symbol>.Empty;
                     hasErrors = true;
                 }
                 else
                 {
-                    member = LookupMemberForPropertyPattern(inputType, name, diagnostics, ref hasErrors, out memberType);
+                    var memberBuilder = ArrayBuilder<Symbol>.GetInstance();
+                    LookupMembersForPropertyPattern(inputType, expr, memberBuilder, diagnostics, ref hasErrors, out memberType);
+                    members = memberBuilder.ToImmutableAndFree();
+                    if (!hasErrors && members.Length > 1)
```

Comment:
>!hasErrors [](http://example.com/codeflow?start=24&length=10)

The check for `!hasErrors` seems unnecessary to me. `hasErrors` may be already set by the caller or from an earlier subpattern. And if we don't report a `/langversion` error now because of another error, won't the user just get the `/langversion` error after fixing the current error?

Reply:
If we don't care if there's already error on the name we might as well move it to the parser. But in that case it will err in completely invalid scenarios `{ (c?a:b): p } // invalid name + feature is in preview (?!)`

Reply:
If we want to avoid reporting this error if the are syntax errors, could we check `!expr.HasErrors` here instead?

Reply:
>hasErrors may be already set by the caller or from an earlier subpattern. And if we don't report a /langversion error now because of another error, won't the user just get the /langversion error after fixing the current error?

Reading this again, that's actually a pretty bad experience. Another scenario is doing `a?.b` which produces "invalid name" and only after fixing that we give langversion error (even with expr.HasError) which is unnecessary back and forth.

I'm leaning towards moving this check to the parser and produce all diagnostics upfront - `(c?a:b)` example is a hypothetical scenario that I think we shouldn't be worry about.

Reply:
> Another scenario is doing a?.b which produces "invalid name" and only after fixing that we give langversion error (even with expr.HasError) which is unnecessary back and forth.
>
> I'm leaning towards moving this check to the parser and produce all diagnostics upfront

Moving this check to the parser would be a fair amount of code churn (with "/langversion" errors reported in parsing tests, etc.). Also, I suspect we have a number of places where we won't produce some binding errors if the expression contains syntax errors. So, to me, it seems sufficient to either remove the `!hasErrors` check above or change to `!expr.HasErrors`.

Reply:
>Also, I suspect there are other codepaths in binding we don't report all binding errors if the expression contains syntax errors. 

Not for subpatterns apparently - probably because we only had an identifier before and it couldn't go wrong. I did a little experiment to see how it looks like (with all green pattern tests; semantic and syntax): https://github.com/dotnet/roslyn/commit/d3f1e7f1bd1c14c2ca3a398eae71a9c18b912a1f?w=1
We already use `preview` version so there wasn't a need to adjust the baseline.

Anywho, if you feel that's gonna be an issue I'll be fine to keep it in the binding.

Reply:
>  I did a little experiment to see how it looks like (with all green pattern tests; semantic and syntax) 

> Anywho, if you feel that's gonna be an issue I'll be fine to keep it in the binder.

My hesitation was just avoiding code churn if there was minimal benefit to moving the check.

It looks like we have a number of pattern feature checks in the parser already so perhaps it makes sense to check this feature in the parser as well.

If you do move the check, please include at least one parsing test with `TestOptions.Regular9` and `TestOptions.RegularPreview`, and consider including a binding test that demonstrates that errors in an earlier subpattern have no effect on the `/langversion` error: perhaps `_ = new C() is { Unrecognized: null, Prop1.Prop2: null };`